xdsl/dialects/snitch_stream.py

```diff
@@ -248,6 +249,110 @@ def __init__(
             },
         )
 
+    def print(self, printer: Printer):
+        with printer.indented():
+            printer.print_string(" {")
+            if self.stride_patterns.data:
+                printer.print_string("\npatterns = [")
+                with printer.indented():
+                    printer.print_list(
+                        self.stride_patterns.data,
+                        lambda attr: printer.print("\n", attr),
+                        delimiter=",",
+                    )
+                printer.print_string("\n]")
+            else:
+                printer.print_string("\npatterns = []")
+        printer.print_string("\n}")
+
+        if self.inputs:
+            printer.print_string(" ins(")
+            printer.print_list(self.inputs, printer.print_ssa_value)
+            printer.print_string(" : ")
+            printer.print_list((i.type for i in self.inputs), printer.print_attribute)
+            printer.print_string(")")
+
+        if self.outputs:
+            printer.print_string(" outs(")
+            printer.print_list(self.outputs, printer.print_ssa_value)
+            printer.print_string(" : ")
+            printer.print_list((o.type for o in self.outputs), printer.print_attribute)
+            printer.print_string(")")
+
+        if self.attributes:
+            printer.print(" attrs = ")
```

Comment:
Would it make sense to change this to just `" attributes "` like `attr-with-keyword` in MLIR?

Reply:
oh yeah